Adds documentation for Remember and Recall

redwizard42 opened the pull request:
Adds documentation on these upcoming features.

Commit:
Adds documentation for Remember and Recall

wescopeland commented on docs/developer-docs/flags/remember.md:
```diff
@@ -0,0 +1,186 @@
+|  7 | Add Hits  | Mem    | 8-Bit  | 0x1000 |  =  | Value |        | 0x03    | 1 (0) |
+|  8 | Measured  | Value  |        | 0x00   |  =  | Value |        | 0x00    | 3 (0) |
+
+In this example, we are using the stored value in multiple chains within an Add Hits chain without having to recalculate the value each time. Here we are looking for specific increases of `0x1234` to occur while the value of `0x1000` is equal to 3.  Perhaps `0x1234` is money, `0x1000` is the current level, and you want to make sure the player collects a gem worth 5 units of money, one worth 20, a ...
```
```suggestion
In this example, we are using the stored value in multiple chains within an Add Hits chain without having to recalculate the value each time. Here we are looking for specific increases of `0x1234` to occur while the value of `0x1000` is equal to 3.  Perhaps `0x1234` is money, `0x1000` is the current level, and you want to make sure the player collects a gem worth 5 units of money, one worth 20, and one worth 100 in level 3. `Recall` is used here to check all these values of the difference between Mem and Delta without having to recalculate it each time. See the [`Add Hits`](/developer-docs/flags/addhits-subhits) document for more information about how that condition works.
```

redwizard42 replied:
Caught that it doesn't have -flag, missed that it doesn't have -and

Fixed

Commit:
Fixes bad link.

wescopeland reviewed (approved):
LGTM. Not sure if it's worth anyone else weighing in. Should this be published now or when the feature ships?

redwizard42 commented:
> LGTM. Not sure if it's worth anyone else weighing in. Should this be published now or when the feature ships?

I don't think there is harm in publishing now, as it includes a note about when they will be available. A benefit would be that if/when a beta is put out (unknown to me if/when this would happen), the documentation will exist for anyone wanting to use it.

wescopeland commented:
Recommend adding a title and description for the new pages. This isn't in the contributing docs so I'll swoop around and get that added later today. eg:

```md
---
title: Understanding and Using AddAddress
description: Learn how to effectively use the AddAddress flag to modify memory addresses for conditions, understand pointer types, calculate offsets, and chain pointers for advanced achievement creation.
---
```

I added a bunch of these to various pages last night (should be on all Flags pages now). This is what appears on Discord embeds and Google search results for the title & description. Description should shoot for <220 characters.

Commit:
Adds title and description metadata.

Jamiras commented on docs/developer-docs/flags/remember.md:
```diff
@@ -0,0 +1,186 @@
+| -- | --------- | ----- | ------ | ------ | --- | ----- | ------ | ------- | ----- |
+|  1 | Remember  | Mem   | 16-bit | 0x1234 |  -  | Delta | 16-Bit | 0x1224  |       |
+
+The Remember condition can optionally take two operand and process a value based on an operator such as multiplication, division, addition, subtraction, and others. In this example, we used the subtraction operator to store the equivalent value as in the first example, but we do it in a single line.
```
two operand _s_

Jamiras commented on docs/developer-docs/flags/remember.md:
```diff
@@ -0,0 +1,186 @@
+| ID | Flag      | Type   | Size   | Memory | Cmp | Type  | Size   | Mem/Val    | Hits  |
+| -- | --------- | ------ | ------ | ------ | --- | ----- | ------ | ---------- | ----- |
+|  1 | Remember  | Mem    | 16-bit | 0x1234 |  -  | Delta | 16-Bit | 0x1224     |       |
+|  2 | Pause If  | Recall |        |        |  >= | Value |        | 0x8000000  | 1 (0) |
```
You're missing a 0 on `0x80000000`

Jamiras commented on docs/developer-docs/flags/remember.md:
```diff
@@ -0,0 +1,186 @@
+|  2 | AddAddress | Mem    | 32-bit | 0x1234 |  +  | Recall |        |         |       |
+|  3 |            | Mem    | 16-bit | 0x08   |  >  | Value  |        | 0x00    |       |
+
+In this example, let's say you have an address that points to a table of data, but in order to know where in the table you want to go, you have to calculate the offset from another source.  `0x1234` will be used as the pointer to the table of data. `0x0789` will be the index in the table. The size of data for table entries is `0x60` bytes. And finally, the data we care about is `0x08` bytes into  ...
```
Missing period between "structure" and "Perhaps"

Jamiras commented on docs/developer-docs/recall.md:
```diff
@@ -0,0 +1,19 @@
+Examples of how it can be used:
+
+- Recall a calculated value for multiple uses in logic.
+- Continuous operation on remembered values to perform maore complicated math than `Add Source` and `Sub Source` can accomplish.
```
"more" is misspelled.

Jamiras reviewed (commented):
Noted a few simple typos that should be corrected, but overall it looks great. I'm very happy that you put in detailed examples of how the logic interacts with PauseIf.

Commit:
Addresses typographical errors identified by Jamiras's PR review.

redwizard42 commented:
> Noted a few simple typos that should be corrected, but overall it looks great. I'm very happy that you put in detailed examples of how the logic interacts with PauseIf.

One of those things that I knew would trip people up sooner or later.

Corrected typos.

Commit:
Fixes minor typo in version availability comments. (period->comma)

Commit:
Adds some additional clarification to the lifetime of a remembered value.

Commit:
Adds link to Recall operand document.

Commit:
Cleans up some additional typos and awkward wording.

redwizard42 commented:
Misunderstanding came up about the lifetime of a remembered value, so I added some additional clarification and also fixed more typos I found as I read through it another two times. I'd like to think this is the final draft.

Commit:
Add in the condition syntax for Remember/Recall

Commit:
Changed the =1 to *2 in the Remember example so that it'd actually be valid example.

wescopeland committed:
chore: run prettier and add i18n links

falsepopsky commented on docs/developer-docs/flags/remember.md:
```diff
@@ -0,0 +1,192 @@
+
+# `Remember`
+
+_This feature will be available at the RA_Integration version 1.4 milestone for developers, with support to be added for rcheevos version 11.4 milestone._
```
Because this is available in a specific version, it's better to highlight it in a custom block.

reference: [Custom Containers](https://vitepress.dev/guide/markdown#custom-containers)

wescopeland replied:
Chatted with @redwizard42 , will be addressed in a follow-up PR.

falsepopsky commented on docs/developer-docs/recall.md:
```diff
@@ -0,0 +1,26 @@
+
+# `Recall`
+
+_This feature will be available at the RA_Integration version 1.4 milestone for developers, with support to be added for rcheevos version 11.4 milestone._
```
same as above

falsepopsky reviewed (commented):
maybe use the custom container `WARNING`, for recall and remember

redwizard42 committed:
Merge branch 'RetroAchievements:main' into remember_recall
